src/meta/src/hummock/manager/compaction/compaction_group_manager.rs

```diff
@@ -280,6 +281,18 @@ impl HummockManager {
         if table_ids.peek().is_none() {
             return Ok(());
         }
+
+        {
+            // Remove table write throughput statistics
+            // The Caller acquires `Send`, so we should safely use `write` lock before the await point.
+            // The table write throughput statistic accepts data inconsistencies (unregister table ids fail), so we can clean it up in advance.
+            let mut table_write_throughput_statistic_manager =
+                self.table_write_throughput_statistic_manager.write();
+            for table_id in table_ids.by_ref() {
```

Comment:
by_ref will advance the iter position and it is an incorrect usage here because L308 requires full iteration on the table_ids.

See the example in by_ref doc [here](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.by_ref)